obspy/imaging/tests/test_mopad_script.py

```diff
@@ -45,14 +47,17 @@ def test_script_convert_type_sdr(self):
 Fault plane 1: strike =  77°, dip =  89°, slip-rake = -141°
 Fault plane 2: strike = 346°, dip =  51°, slip-rake =   -1°
 '''
-        expected = expected.encode('utf-8')
 
         result = out.stdout[:-1]
-        if result.startswith(b'b'):
-            # Total mojibake when the system is configured strangely. Mostly
-            # only occurs when building packages, so that LANG=C, and Python
-            # picks ASCII for output encoding.
-            expected = str(expected).encode('ascii')
+        try:
+            expected = expected.encode(sys.stdout.encoding)
+        except:
+            expected = expected.encode('utf-8')
+            if PY3:
+                # This may look like Py3k does extra stuff, but it's actually
+                # saner!
+                expected = repr(expected)
+                expected = expected.encode('ascii')
```

Comment:
@QuLogic I just do not get the idea. If I enter the `if PY3`, `expected` is a byte string, the `repr` of a byte str is normal string containing something like `"b'blah_blub'"`. Then encoding this normal string results in the byte string `b"b'blah_blub'"` which is different from a simple byte string obtained via `out.stdout[:-1]`. What am I missing?